src/components/todos/todo-list-table.tsx

```diff
@@ -57,49 +62,70 @@ const TodoListTableRow = ({
   userId,
 }: TodoListTableRowProps) => {
   const isEditTodoVisible = todo.assignee?.assignee_id === userId
-  return (
-    <TableRow>
-      <TableCell className="whitespace-nowrap">{todo.title}</TableCell>
+  const [showViewTodoModal, setShowViewTodoModal] = useState(false)
 
-      <TableCell className="whitespace-nowrap">
-        <TodoStatusTable status={todo.status} />
-      </TableCell>
+  const { mutation, handleSubmission } = useUpdateTask({
+    todo,
+  })
 
-      <TableCell className="whitespace-nowrap">
-        <TodoLabelsList labels={todo.labels ?? []} />
-      </TableCell>
+  const handleSubmit = (todoDetails: TTodoFormData) => {
+    handleSubmission(todoDetails, () => {
+      setShowViewTodoModal(false)
+    })
+  }
 
-      <TableCell className="whitespace-nowrap">
-        {todo.priority ? <TaskPriorityLabel priority={todo.priority} /> : '--'}
-      </TableCell>
+  return (
+    <TodoDialog
+      mode={isEditTodoVisible ? 'edit' : 'view'}
+      defaultData={TodoUtil.getDefaultTodoFormData(todo)}
+      onOpenChange={setShowViewTodoModal}
+      open={showViewTodoModal}
+      onSubmit={handleSubmit}
+      isMutationPending={mutation.isPending}
+    >
```

Comment:
- as discussed instead of wrapping the entire row in `TodoDialog` open the dialog only when the user clicks on the todo title.